dvc/fs/gs.py

```diff
@@ -1,228 +1,43 @@
-
-
-class GSFileSystem(BaseFileSystem):
+class GSFileSystem(FSSpecWrapper):
     scheme = Schemes.GS
     PATH_CLS = CloudURLInfo
-    REQUIRES = {"google-cloud-storage": "google.cloud.storage"}
+    REQUIRES = {"gcsfs": "gcsfs"}
     PARAM_CHECKSUM = "md5"
     DETAIL_FIELDS = frozenset(("md5", "size"))
 
     def __init__(self, repo, config):
         super().__init__(repo, config)
 
-        url = config.get("url", "gs:///")
+        url = config.get("url", "gs://")
         self.path_info = self.PATH_CLS(url)
 
-        self.projectname = config.get("projectname", None)
-        self.credentialpath = config.get("credentialpath")
-
-    @wrap_prop(threading.Lock())
-    @cached_property
-    def gs(self):
-        from google.cloud.storage import Client
-
-        return (
-            Client.from_service_account_json(self.credentialpath)
-            if self.credentialpath
-            else Client(self.projectname)
-        )
-
-    def _generate_download_url(self, path_info, expires=3600):
-        import google.auth
-        from google.auth import compute_engine
-
-        expiration = timedelta(seconds=int(expires))
-
-        bucket = self.gs.bucket(path_info.bucket)
-        blob = bucket.get_blob(path_info.path)
-        if blob is None:
-            raise FileNotFoundError
-
-        if isinstance(
-            blob.client._credentials,  # pylint: disable=protected-access
-            google.auth.credentials.Signing,
-        ):
-            # sign if we're able to sign with credentials.
-            return blob.generate_signed_url(expiration=expiration)
-
-        auth_request = google.auth.transport.requests.Request()
-        # create signing credentials with the default credentials
-        # for use with Compute Engine and other environments where
-        # Client credentials cannot sign.
-        signing_credentials = compute_engine.IDTokenCredentials(
-            auth_request, ""
-        )
-        return signing_credentials.signer.sign(blob)
-
-    def exists(self, path_info, use_dvcignore=True):
-        """Check if the blob exists. If it does not exist,
-        it could be a part of a directory path.
-
-        eg: if `data/file.txt` exists, check for `data` should return True
-        """
-        return self.isfile(path_info) or self.isdir(path_info)
+        self.login_info = self._prepare_credentials(config)
 
-    def isdir(self, path_info):
-        dir_path = path_info / ""
-        return bool(list(self._list_paths(dir_path, max_items=1)))
+    def _prepare_credentials(self, config):
+        login_info = {}
+        login_info["project"] = config.get("projectname")
+        login_info["token"] = config.get("credentialpath")
+        return login_info
 
-    def isfile(self, path_info):
-        if path_info.path.endswith("/"):
-            return False
+    def _entry_hook(self, entry):
+        if "md5Hash" in entry:
+            entry["md5"] = base64.b64decode(entry["md5Hash"]).hex()
+        return entry
```

Comment:
I guess no longer be needed if we decide to switch to etag https://github.com/iterative/dvc/issues/5527 Out of scope for this PR though.

Reply:
yep. This is temporary.